raster-foundry/raster-foundry
# Add hotkeys for annotation-related actions. Enable shape rotate/rescale.

## Overview

This PR adds a bunch of hotkeys for annotation create/edit/clone, metadata edit/submit/cancel, annotation shape rotate/rescale and move north/south/east/west.  A detailed list of available keyboard shortcuts are provided on the UI and also listed in the screenshot under the demo section.  GIFs showing the use of these actions using keyboard is under the testing instruction section.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo
<img width="1680" alt="hotkey list" src="https://github.com/aaronxsu/MyData/raw/master/gifs/rf-annotations/hotkey_list.png">


### Notes

Thanks @designmatty for the cool new icons!

## Testing Instructions

 * Go to `/projects/edit/:project_id/annotate` or click on "Annotate project" button on the project edit page
 * See shortcut tips
![tips](https://github.com/aaronxsu/MyData/raw/master/gifs/rf-annotations/tips.gif)

 * Create a new rectangular annotation with `shift+alt+r`, label name field should autofocus, use `shift+enter` to submit
![create](https://github.com/aaronxsu/MyData/raw/master/gifs/rf-annotations/create.gif)

 * Click an annotation on the sidebar or on the map, use `shift+ctrl+e` to start editing, use `shift+ctrl+r` to rotate and rescale the shape, use `shift+enter` to submit changes, use `shift+esc` to cancel
![edit and rotate](https://github.com/aaronxsu/MyData/raw/master/gifs/rf-annotations/edit_and_rotate.gif)
![cancel edit](https://github.com/aaronxsu/MyData/raw/master/gifs/rf-annotations/cancel_edit.gif)

 * Click an annotation on the sidebar or on the map, use `shift+ctrl+c` to start cloning, use `shift+alt+up/down/left/right` to move annotations around
![clone and move](https://github.com/aaronxsu/MyData/raw/master/gifs/rf-annotations/clone_and_move.gif)

 * Click an annotation on the sidebar or on the map, use `shift+ctrl+backspace` to delete
![delete](https://github.com/aaronxsu/MyData/raw/master/gifs/rf-annotations/delete.gif)
